src/index.ts

```diff
@@ -1,61 +1,111 @@
 import { Reshuffle, BaseConnector, EventConfiguration } from 'reshuffle-base-connector'
+import Shopify, { WebhookTopic } from 'shopify-api-node'
+import { Request, Response } from 'express'
 
-export interface _CONNECTOR_NAME_ConnectorConfigOptions {
-  var1: string
-  // ...
+const DEFAULT_WEBHOOK_PATH = '/reshuffle-shopify-connector/webhook'
+
+export interface ShopifyConnectorConfigOptions extends Shopify.IPrivateShopifyConfig {
+  baseURL?: string
+  webhookPath?: string
+  webhookName?: string
 }
 
-export interface _CONNECTOR_NAME_ConnectorEventOptions {
-  option1?: string
-  // ...
+export interface ShopifyConnectorEventOptions {
+  topic: WebhookTopic
 }
 
-export default class _CONNECTOR_NAME_Connector extends BaseConnector<
-  _CONNECTOR_NAME_ConnectorConfigOptions,
-  _CONNECTOR_NAME_ConnectorEventOptions
+function validateBaseURL(url?: string): string {
+  if (typeof url !== 'string') {
+    throw new Error(`Invalid url: ${url}`)
+  }
+  const match = url.match(/^(https:\/\/[\w-]+(\.[\w-]+)*(:\d{1,5})?)\/?$/)
+  if (!match) {
+    throw new Error(`Invalid url: ${url}`)
+  }
+  return match[1]
+}
+
```

Comment:
Not for this PR - but is it about time we moved this function out of connector code and into reshuffle?

Reply:
exactly right! I thought about doing it today actually.
I'd say into reshuffle-base-connector, as a connector doesn't import Reshuffle